src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
```

Comment:
i'm fine with whatever but it's going to look super ugly because it's the caller of `get1key` and `get2keys` and `get2keys` etc who needs to have a catch. That looks awful to me (and the `@SuppressWarnings` in the callee side are still be necessary)

```
        try {
            final RecalDatum existingDatum = table.get2Keys(key0, key1);
        } catch(ClassCastException e){
            throw new ...
        }

```